src/compiler/checker.ts

```diff
@@ -10403,6 +10433,7 @@ namespace ts {
                 case SyntaxKind.TemplateExpression:
                     return checkTemplateExpression(<TemplateExpression>node);
                 case SyntaxKind.StringLiteral:
+                    return checkStringLiteralExpression(<LiteralExpression>node);
                 case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
@tinganho Not that I know of, the VM techniques generally are called ```string inlining```, ```const folding```, not sure what Chrome & al. do here:
http://hhvm.com/blog/713/hhvm-optimization-tips

Not even the model above performs particularly well in Chrome looking at the generated code. Maybe Chakra does more ```voodoo```. Could be better with WebAssembly :smile: